nautobot/core/utils/filtering.py

```diff
@@ -150,11 +150,7 @@ def get_filterset_parameter_form_field(model, parameter, filterset=None):
                 choices_as_strings=True, queryset=queryset_map[plural_name]().as_queryset()
             )
     elif isinstance(field, (MultipleChoiceFilter, ChoiceFilter)) and "choices" in field.extra:
-        form_field_class = forms.ChoiceField
-        form_field_class.widget = StaticSelect2Multiple()
-        form_attr = {"choices": field.extra.get("choices")}
-
-        form_field = form_field_class(**form_attr)
+        form_field = forms.MultipleChoiceField(choices=field.extra.get("choices"), widget=StaticSelect2Multiple)
```

Comment:
In the [original fix](https://github.com/nautobot/nautobot/commit/4175e8b6ab4592c4ddd249ac1a7f5b1b43038c7e) a similar change was made to an `elif isinstance(field, BooleanFilter)` check that seems to not even exist in 2.0. Would you mind for my curiosity chasing down why that check is no longer present/needed?

Reply:
It was removed completely in https://github.com/nautobot/nautobot/pull/3479/files#diff-56c997d3b0c57888c53b79e77be7d85c6302706fbf9f829aba3ad3bd2cd373d5

I assume because the default behavior was determined to be the desired behavior? We don't have many BooleanFilters but I can try to add a test for one.

Reply:
Looks like that one is broken. When you change values in the "has local config context data" filter in the device list view you get this JS error in the console. It looks like our form code is expecting everything to use select2 but I can't figure out where that's happening.

![image](https://github.com/nautobot/nautobot/assets/75227981/f4cc94a0-2029-465a-9878-c374bcb3aa84)

Reply:
Could be [here](https://github.com/nautobot/nautobot/blob/cafab53d861cc625cdde9da11121895d7e94a8a0/nautobot/project-static/js/forms.js#L539)

Reply:
Darn that @glennmatthews, always causing problems...

Reply:
The original fix had a ChoiceField for booleans but NullBooleanField looks a little more flexible with the values it accepts. It doesn't seem to make a difference in this context though, so should we stick with ChoiceField for consistency with ltm-1.6?

Reply:
I *think* ChoiceField rather than NullBooleanField is appropriate in the filter context, as I *think* the user shouldn't be able to set a filter of `has_local_config_context_data=null`?